<launch>

    <arg name="model" default="burger" doc="model type [burger, waffle]"/>
    <arg name="x_pos" default="-5.0"/>
    <arg name="y_pos" default="4.0"/>
    <arg name="z_pos" default="0.0"/>
    <arg name="yaw_angle" default="3.14"/>

    <rosparam command="load" file="$(find turtlebot3_social)/param/forceFactors.yaml" />
    <arg name="world_file"  default="$(find turtlebot3_RL)/worlds/house_ped.world"/>
    <arg name="gui" default="false"/>
    <arg name="rviz" default="false"/>
    <arg name="use_sim_time" default="true"/>
    <arg name="save" default="false"/>
    <!-- set use_sim_time flag -->
    <group if="$(arg use_sim_time)">
        <param name="/use_sim_time" value="true" />
    </group>

    <node name="gazebo" pkg="gazebo_ros" type="gzserver" output="screen"
        args="$(arg world_file)" />

    <group if="$(arg rviz)">
        <node name="rviz" pkg="rviz" type="rviz" args="-d $(find actor_services)/rviz/tf_show.rviz"/>
    </group>

    <group if="$(arg gui)">
        <node name="gazebo_gui2" pkg="gazebo_ros" type="gzclient" respawn="false" output="screen"/>
    </group>

    <param name="x_pos"  value="$(arg x_pos)"/>
    <param name="y_pos"  value="$(arg y_pos)"/>
    <param name="z_pos"  value="$(arg z_pos)"/>
    <param name="yaw_angle"  value="$(arg yaw_angle)"/>

    <param name="robot_description" command="$(find xacro)/xacro.py $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />

    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model turtlebot3_burger -x $(arg x_pos) -y $(arg y_pos) -z $(arg z_pos) -Y $(arg yaw_angle) -param robot_description" />


</launch>